### PR TITLE
azurerm_search_service:Update basic SKU partition count from 1 to 3

### PR DESCRIPTION
**Pull request by Gnana-Bharathi-K**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @Gnana-Bharathi-K - this looks good to me, but could we include a basic sku with partition count set to 3 in the tests so we can confirm this works as expected? Thanks!

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m Thanks for reviewing I have updated the basic SKU partition count in tests to 3 from 4 kindly please review the tests.

File: internal/services/search/search_service_resource_test.go [hunk 1]
```diff
@@ -278,7 +278,7 @@ func TestAccSearchService_partitionCountInvalidBySku(t *testing.T) {
 		{
 			Config:      r.partitionCount(data, "basic", 3),
```
Comment (catriona-m) [inline internal/services/search/search_service_resource_test.go:279]: should we change this to 4 since 3 is now a valid partition count?

Comment (Gnana-Bharathi-K) [reply internal/services/search/search_service_resource_test.go:279]: Yes will update and the initial count I mentioned it to 4 also I have tested both cases successfully with local provider 
1. created resource with basic SKU partition count 3
2. with partition count 4 I got error message  "Error: 'partition_count' values greater than 3 cannot be set for the "basic" SKU, got 4)"

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m The Test Results for your reference
image

Please let me know if further details required

**Comment by catriona-m** [comment]
> Hi @catriona-m The Test Results for your reference image
> 
> Please let me know if further details required

Thanks for updating the existing test @Gnana-Bharathi-K. 

Could you also add a new test using the partitionCount config using a basic sku and a partition count of 3 so that we can confirm that it is possible to create an `azurerm_search_service` with a basic sku and a partition count of 3.

Thank you!

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m executed the test function with partition count 3 and SKU as basic, the search service successfully deployed in the azure portal. 
Resource Group Name "acctestRG-search-241126180102558706"
search service name "acctestsearchservice241126180102558706"

I have added separate combined test function "TestAccSearchService_partitionCountvalidBySkuBasic" to test and let me know if we need separate test function only for deployment or combined test function.

Result when I passed basic sku with partition count 4
image

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m , please let me know is there something I need to perform !

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for updating this @Gnana-Bharathi-K - it looks like there's a linting issue, could you run `make fmt` on this to fix it? Thanks!

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m updated the format of the file search_service_resource_test.go

**Comment by Gnana-Bharathi-K** [comment]
@catriona-m  Updated the lint format with running gofmt against the file search_service_resource_test.go there was an extra line space which was resolved.

**Comment by Gnana-Bharathi-K** [comment]
I think all the test cases been passed successfully, is there something we still need to review @catriona-m ?

**Comment by Gnana-Bharathi-K** [comment]
@catriona-m please let me know if the there is any changes need to be done by my end thanks !

**Review by catriona-m — APPROVED** [review]
Thanks for updating this @Gnana-Bharathi-K - LGTM!

**Comment by Gnana-Bharathi-K** [comment]
Hi @catriona-m thanks for updates and I guess everything went well ! Let me know if something needs to addressed thanks for the collaboration!

**Pull request merged**